system_modules/napvideo/src/video.cpp

```diff
@@ -789,12 +800,25 @@ namespace nap
 			else if (packet == mSeekEndPacket.get())
 			{
 				VIDEO_DEBUG_LOG("seek end received");
-
-				clearFrameQueue(mSeekFrameQueue, false);
+                {
+                    // clear any previous frames, we're not interested in them anymore
+                    clearFrameQueue(mFrameQueue, false);
+
+                    // Copy seek frame queues into the regular frame queue
+                    while(!mSeekFrameQueue.empty())
+                    {
+                        Frame f = popSeekFrame();
+                        std::unique_lock<std::mutex> lock(mFrameQueueMutex);
+                        mFrameQueue.push(f);
```

Comment:
Why lock and unlock the thread every iteration? Probably better to do that before.

Reply:
How is this resolved exactly? You modify the `mFrameQueue` without a lock now?

Reply:
Indeed swapping is better, changed it -> https://github.com/napframework/nap/pull/28/commits/9ac756a628c02532b8949007f6f62590580367f4